### ISSUE TITLE
is this hooks automatically disconnect channels?

### ISSUE THREAD

**Issue by 201524547** [mayteio/use-pusher#40]
useChannel hooks automatically disconnect channels?

i use this hooks, i wonder useChannel hook automatically disconnect when component unmount?

**Comment by mayteio** [comment]
Thanks for letting me know. Do you have a codesandbox that reproduces this?

**Comment by underfisk** [comment]
@201524547 It was something i had reported time ago, it disconnects and never re-connects back to the channel if the component re-renders

**Comment by mayteio** [comment]
Please provide a codesandbox - I do not have this issue.

**Comment by underfisk** [comment]
Sure i'll do that but i think i found why its probably happening, let me explain it quickly
There are types of channels that need to be persisted while the application is on, usually those are bounded to the user itself and not to the component, useChannel makes it unsubscribe on unmount and what if a component uses a user room that needs to be alive while the app is running? That was in fact one of the issues, my workaround was creating a custom useChannel that just simply doesn't unsubscribe but removes the event only but i think there is room for improvement.
Do you think that considering a parameter option whether we can unsubscribe or not? It does make sense having it unsubscribing on unmount but sometimes you don't want that channel to go away in sub-components, not sure what's your suggestion in here. Once i have some free time i'll publish here the issue

**Comment by mayteio** [comment]
Hey, why don’t you move the useChannel hook higher up into a provider? That way it can live connected for the duration of the session.

**Comment by underfisk** [comment]
At some point if my event uses that channel somehow its unsubscribing, i tried that but i can check again
What i ended up doing is check if the channel exists and re-use but never close it (only the higher up provider) is closing it
but probably i might be doing something wrong

**Comment by mayteio** [comment]
Might #43 solve this too?

**Comment by jan-wilhelm** [comment]
yeah, I ran into the same problem and (as far as I can tell), #43 would solve this perfectly.
